### ISSUE TITLE
Rocket username and channel are mandatory but not documented

### ISSUE THREAD

**Issue by GJKrupa** [fluxcd/notification-controller#594]
The documentation for the Rocket notification provider says the following:

> To configure a Provider for Rocket, create a Secret with [the address](https://fluxcd.io/flux/components/notification/provider/#address-example) set to the Rocket [webhook URL](https://docs.rocket.chat/guides/administration/admin-panel/integrations#incoming-webhook-script), and a rocket Provider with a [Secret reference](https://fluxcd.io/flux/components/notification/provider/#secret-reference).

However, the provider code checks for the presence of both username and channel properties as well and will error leaving it in a not ready state if these are not provided:
```
	if username == "" {
		return nil, errors.New("empty Rocket username")
	}

	if channel == "" {
		return nil, errors.New("empty Rocket channel")
	}
```

**Comment by GJKrupa** [comment]
If a webhook URL is used, neither the username not the channel should be required since these are both tied to the webhook URL in the Rocket workspace configuration.

**Comment by somtochiama** [comment]
@GJKrupa You are right, from the rocket documentation, the channel and username are not required.
Do you want to raise a PR to fix this. We can take out the checks in the `NewRocket` function and only set the channel/username in the slack payload if they are provided

**Comment by GJKrupa** [comment]
https://github.com/fluxcd/notification-controller/pull/603 raised
